recourse_methods/dice_method.py

```diff
@@ -0,0 +1,108 @@
+from typing import Sequence, Optional, Mapping, Any
```

Comment:
High-level comment: please consider adding function docstrings with `Args` and `Returns` sections (unless the function is very basic function and seems like it will be self-explanatory to a reader).